### ISSUE TITLE
SAM log file  - flush

### ISSUE THREAD

**Issue by dsrini-open** [aws/aws-sam-cli#1985]
### Description
SAM with log file option -l does output to log file, but logs only found after stopping the SAM command. 

### Steps to reproduce
sam local start-api -l <log_file>

### Observed result
The output log file is generated only after the sam command finishes. The log could not be accessed with tail as, there is no output to the log file. 

### Expected result
Log file generated and flushed periodically to access the log.

### Additional environment details (Ex: Windows, Mac, Amazon Linux etc)
1. OS:CentOS 7
2. `sam --version`: 0.48

**Comment by jfuss** [comment]
@dsrini-open Curious on the use-case to use it like this? Why do you want to tail the file instead of using the console output that is already doing what you seem to want.

**Comment by dsrini-open** [comment]
@jfuss Then, I don't see how -l flag is useful ? 
When using -l flag (logging to a file), I do not log outputs in the command line output. And, I don't see anything on the log file.

**Comment by jfuss** [comment]
@dsrini-open `-l` was meant to dump the logs to a file after `invoke`, so they can be easily saved. This rolled up into `start-api` and `start-lambda`. Tailing happens by default in the console output, which is why I asked your use-case for using `-l` instead of just reading the output from the command directly.

I am not sure what it takes to frequently flush but could be a good improvement.

**Comment by solarmosaic-kflorence** [comment]
@jfuss Hm, when I invoke my lambda locally (which is build with a container), I only seem to get all my logs output to console when the lambda method finishes. Is that expected? Perhaps a separate, but potentially related, issue.

**Comment by kstro21** [comment]
@jfuss the console output sometimes is unreadable and unformatted. Dumping to a file makes them easy to read.
You can have an idea with the attached SS where I can only see the full error in the log file because the console output is not helpful at all. But I had to Ctrl+C the `sam local start-api` process.

![image](https://user-images.githubusercontent.com/5972646/117896008-a13a0880-b28d-11eb-9ade-3463cea15094.png)

**Comment by jfuss** [comment]
This was already patched and released and therefore closing. Please re-open or create a new issue if you are still experiencing issues.
